src/cobald/daemon/runners/thread_runner.py

```diff
@@ -1,81 +1,55 @@
 import threading
-import time
+import asyncio
 
-from ..debug import NameRepr
 from .base_runner import BaseRunner, OrphanedReturn
 
 
-class CapturingThread(threading.Thread):
-    """
-    Daemonic threads that capture any return value or exception from their ``target``
+class ThreadRunner(BaseRunner):
     """
+    Runner for subroutines with :py:mod:`threading`
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs, daemon=True)
-        self._exception = None
-        self._name = str(NameRepr(self._target))
-
-    def join(self, timeout=None):
-        super().join(timeout=timeout)
-        if self._started.is_set() and not self.is_alive():
-            if self._exception is not None:
-                raise self._exception
-        return not self.is_alive()
-
-    def run(self):
-        """Modified ``run`` that captures return value and exceptions from ``target``"""
-        try:
-            if self._target:
-                return_value = self._target(*self._args, **self._kwargs)
-                if return_value is not None:
-                    self._exception = OrphanedReturn(self, return_value)
-        except BaseException as err:
-            self._exception = err
-        finally:
-            # Avoid a refcycle if the thread is running a function with
-            # an argument that has a member that points to the thread.
-            del self._target, self._args, self._kwargs
-
-
-class ThreadRunner(BaseRunner):
-    """Runner for subroutines with :py:mod:`threading`"""
+    All active payloads are *not* cancelled when the runner is closed.
+    Only program termination forcefully cancels leftover payloads.
+    """
 
     flavour = threading
 
-    def __init__(self):
-        super().__init__()
-        self._threads = set()
+    # This runner directly uses threading.Thread to run payloads.
+    # To detect errors, each payload is wrapped; errors and unexpected return values
+    # are pushed to a queue from which the main task re-raises.
+    def __init__(self, asyncio_loop: asyncio.AbstractEventLoop):
+        super().__init__(asyncio_loop)
+        self._failure_queue = asyncio.Queue()
+
+    def register_payload(self, payload):
+        thread = threading.Thread(
+            target=self._monitor_payload, args=(payload,), daemon=True
+        )
+        thread.start()
 
     def run_payload(self, payload):
         # - run_payload has to block until payload is done
         # instead of running payload in a thread and blocking this one,
         # we just block this thread by running payload directly
```

Comment:
The comment style is somehow different and I don't get what the hyphen is about.

```suggestion
        # The method has to block until payload is done.
        # Instead of running payload in a thread and blocking this one,
        # this thread is blocked by running the payload directly.
```